src/os/bluestore/KernelDevice.cc

```diff
@@ -152,7 +152,9 @@ int KernelDevice::open(const string& p)
   size &= ~(block_size - 1);
 
   r = _aio_start();
-  assert(r == 0);
+  if (r < 0) {
```

Comment:
as close() wouldn't be called after this failure one should delete/reset fs as well.